mpc-core/src/protocols/rep3_ring/ring/int_ring.rs

```diff
@@ -38,18 +43,36 @@ pub trait IntRing2k:
     + Sized
     + Send
     + Sync
+    + TryFrom<u128, Error: Debug>
+    + TryFrom<u64, Error: Debug>
+    + PartialEq
+    + PartialOrd
     + 'static
 {
+    /// Specifies the signed version of this type
     type Signed: Neg<Output = Self::Signed> + From<bool> + AsPrimitive<Self>;
+
+    /// Specifies the number of bits in this type
     const K: usize;
+
+    /// Specifies the number of bytes used for storage in this type
     const BYTES: usize;
 
+    /// Reads a value of this type from a reader
     fn from_reader<R: std::io::Read>(reader: R) -> IoResult<Self>;
+
+    /// Writes a value of this type to a writer
     fn write<W: std::io::Write>(&self, writer: W) -> IoResult<()>;
 
     /// Returns the effective number of bits (i.e., how many LSBs are set)
     fn bits(&self) -> usize;
 
+    /// Casts this type to a BigUint
+    fn cast_to_biguint(&self) -> BigUint;
+
+    /// Casts a BigUint to this type
+    fn cast_from_biguint(biguint: &BigUint) -> Self;
```

Comment:
what happens when this is larger than the type? should this not return a result?

Reply:
is really meant as a downcast that cuts off stuff

Reply:
maybe add a comment then

Reply:
done